garageScript/c0d3-app
# Replace deleteExercise with removeExercise resolver in the client

## Motivation
After adding the [removeExercise](https://github.com/garageScript/c0d3-app/pull/2657) resolver that aims at correcting our initial implementation for removing exercises, we need to use it in the client-side now.

## Steps taken
1. Replace `deleteExercise` with `removeExercise` in mentor page
2. Replace `deleteExercise` with `removeExercise` in admin/lessons page
3. Replace `deleteExercise` with `removeExercise` in `AdminLessonExerciseCard` component
4. Replace `deleteExercise` with `removeExercise` in exercises page
5. Update the tests queries mock to reflect the new changes
6. Prevent removed exercises from showing to the students

## Testing
1. Go to `/exercises/js0`
2. Click `SOLVE EXERCISES`
3. Flag the exercise
4. Go to `/admin/lessons/js0/exercises`
5. Remove the exercise by clicking the `Remove` button
6. Refresh the page and it shouldn't be displayed
7. Go to `/exercises/js0` again
8. Notice how the removed exercise isn't appear
9. Go to `/curriculum/js0/mentor`
10. Click any exercise dropdown button and click `Delete` button to remove the exercise
11. Go to `/exercises/js0` and notice how the exercise isn't showing
